spec/requests/chatops_controller_authentication_spec.rb

```diff
@@ -0,0 +1,61 @@
+require "rails_helper"
+
+class HmacChatopsController < ApplicationChatopsController
+  def index
+    render json: {is_ok: "Sure. Why not?"}
+  end
+end
+
+RSpec.describe "ApplicationChatopsController authentication", type: :request do
```

Comment:
`chatops_controller_authentication_spec` is testing the base class (`ApplicationChatopsController`) behavior for HMAC verification. It wants to confirm that when any controller inherits from this class, HMAC verification is required order for any action to be performed.

This is separated out from the actual `Slack::SlashCommandController` because the same base class will be used by other Slack API handlers such as a controller that watches for a user to be added to a channel and sends a welcome. It's the equivalent of putting user auth in `ApplicationController`.

This could have been used `Slack::SlashCommandController` but that class didn't exist when I wrote this part and I kind of like testing`ApplicationChatopsController` with a bit more isolation.

Reply:
I think that makes sense, although it may end up making sense to abstract the authentication testing into a rspec shared example as we go. Seems like we'd need to set everything up similarly for each set of specs.